packages/valory/skills/decision_maker_abci/states/base.py

```diff
@@ -150,6 +150,17 @@ def bet_amount(self) -> int:
         """Get the calculated bet amount."""
         return int(self.db.get_strict("bet_amount"))
 
+    @property
+    def weighted_accuracy(self) -> int:
+        """Get the weighted accuracy of the selected tool."""
+        tool_name = self.mech_tool
+        store_tools = list(self.policy.weighted_accuracy.keys())
+
+        if tool_name not in store_tools:
+            return None
```

Comment:
The upstream does not expect this. However, I would suggest raising instead as this should never happen. That would be an implementation error.

Reply:
Fixed